packages/fxa-auth-server/lib/routes/cloud-tasks.ts

```diff
@@ -4,113 +4,44 @@
 
 import isA from 'joi';
 import { Container } from 'typedi';
+
 import { ConfigType } from '../../config';
 import DESCRIPTION from '../../docs/swagger/shared/descriptions';
-import { AccountDeleteManager, ReasonForDeletion } from '../account-delete';
-import DB from '../db/index';
+import {
+  AccountDeleteManager,
+  ReasonForDeletion,
+  ReasonForDeletionValues,
+} from '../account-delete';
 import { AuthLogger, AuthRequest } from '../types';
 import validators from './validators';
-import { ERRNO } from '../error';
-import pushBuilder from '../push';
 
 export type DeleteAccountTaskPayload = {
   uid: string;
   customerId?: string;
   reason: ReasonForDeletion;
 };
-type FxaDbForCloudTask = Pick<
-  Awaited<ReturnType<ReturnType<typeof DB>['connect']>>,
-  'account' | 'devices'
->;
-type PushForCloudTask = Pick<
-  ReturnType<typeof pushBuilder>,
-  'notifyAccountDestroyed'
->;
-type Log = AuthLogger & { activityEvent: (data: Record<string, any>) => void };
 
 export class CloudTaskHandler {
-  private log: Log;
-  private db: FxaDbForCloudTask;
-  private push: PushForCloudTask;
   private accountDeleteManager: AccountDeleteManager;
 
-  constructor({
-    log,
-    db,
-    push,
-  }: {
-    log: Log;
-    db: FxaDbForCloudTask;
-    push: PushForCloudTask;
-  }) {
-    this.log = log;
-    this.db = db;
-    this.push = push;
-
+  constructor(private log: AuthLogger) {
     this.accountDeleteManager = Container.get(AccountDeleteManager);
   }
 
   async deleteAccount(taskPayload: DeleteAccountTaskPayload) {
-    try {
-      // get account from MySQL
-      const account = await this.db.account(taskPayload.uid);
-
-      // We fetch the devices to notify before deleteAccount()
-      // because obviously we can't retrieve the devices list after!
-      const devices = await this.db.devices(taskPayload.uid);
-      const push = this.push;
-      const log = this.log;
-
-      const notify = async () => {
-        try {
-          log.info('accountDeleted.byCloudTask', account);
-          await push.notifyAccountDestroyed(taskPayload.uid, devices);
-          await log.notifyAttachedServices('delete', {} as AuthRequest, {
-            uid: taskPayload.uid,
-          });
-          // because a cloud task request is very different from a user request
-          // we cannot emit metrics in the same fashion
-          log.activityEvent({ uid: account.uid, event: 'account.deleted' });
```

Comment:
I think this was lost in the shuffle.  In a user requested delete, the activity event is emitted as part of the `emitMetricsEvent` call because the event name matches https://github.com/mozilla/fxa/blob/fdfb840db9c76d69e93554c4db1c200fc6152afd/packages/fxa-auth-server/lib/metrics/events.js#L13

Reply:
I moved the activityEvent to the account handler where it originally was.

Reply:
Sorry about the confusion.  This event is specifically for delete by cloud tasks.  The /account/destroy handler has an `emitMetricsEvent` call that will emit the activity event.

Reply:
Removing that event would reintroduce https://mozilla-hub.atlassian.net/browse/FXA-8999 into our deletes.